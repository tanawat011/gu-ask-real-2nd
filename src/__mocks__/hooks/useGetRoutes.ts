export const main = {
  fullPath: '/',
  _: {
    home: {
      fullPath: '/',
      _: {
        dashboard: {
          fullPath: '/',
        },
      },
    },
    app: {
      fullPath: '/',
      _: {
        todo: {
          fullPath: '/',
          _: {
            dashboard: {
              fullPath: '/',
            },
          },
        },
        article: {
          fullPath: '/',
        },
      },
    },
    game: {
      fullPath: '/',
      _: {
        sudoku: {
          fullPath: '/',
        },
        ticTacToe: {
          fullPath: '/',
        },
      },
    },
    uiComponent: {
      fullPath: '/',
      _: {
        input: {
          fullPath: '/',
          _: {
            button: {
              fullPath: '/',
            },
            buttonGroup: {
              fullPath: '/',
            },
            calendar: {
              fullPath: '/',
            },
            checkbox: {
              fullPath: '/',
            },
            dateAndTime: {
              fullPath: '/',
            },
            radioButton: {
              fullPath: '/',
            },
            rating: {
              fullPath: '/',
            },
            select: {
              fullPath: '/',
            },
            slider: {
              fullPath: '/',
            },
            switch: {
              fullPath: '/',
            },
            textField: {
              fullPath: '/',
            },
            upload: {
              fullPath: '/',
            },
          },
        },
      },
    },
  },
}

export const auth = {
  fullPath: '/auth',
  _: {
    login: {
      fullPath: '/',
    },
    register: {
      fullPath: '/',
    },
    resetPassword: {
      fullPath: '/',
    },
    forgotPassword: {
      fullPath: '/',
    },
  },
}

export const error = {
  fullPath: '/',
  _: {
    404: {
      fullPath: '/',
    },
    401: {
      fullPath: '/',
    },
    403: {
      fullPath: '/',
    },
    500: {
      fullPath: '/',
    },
    501: {
      fullPath: '/',
    },
    502: {
      fullPath: '/',
    },
    503: {
      fullPath: '/',
    },
    504: {
      fullPath: '/',
    },
  },
}

const routers = {
  main,
  auth,
  error,
}

export const useGetRoutes = () => {
  const all = routers

  return { all, ...routers }
}
