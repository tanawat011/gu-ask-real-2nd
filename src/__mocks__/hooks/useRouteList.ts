export const authRoutes = {
  fullPath: '/auth',
  children: {
    login: {
      fullPath: '/auth/login',
    },
  },
}

export const componentRoutes = {
  children: {
    card: {
      fullPath: '/',
    },
    button: {
      fullPath: '/',
    },
    input: {
      fullPath: '/',
    },
  },
}

export const mainRoutes = {
  children: {
    home: {
      fullPath: '/',
    },
    todo: {
      fullPath: '/',
    },
    article: {
      fullPath: '/',
    },
    component: {
      fullPath: '/',
      ...componentRoutes,
    },
  },
}

export const errorRoutes = {
  children: {
    e401: {
      fullPath: '/',
    },
    e403: {
      fullPath: '/',
    },
    e404: {
      fullPath: '/',
    },
    e500: {
      fullPath: '/',
    },
    e502: {
      fullPath: '/',
    },
    e503: {
      fullPath: '/',
    },
    e504: {
      fullPath: '/',
    },
  },
}

export const allRoutes = {
  auth: authRoutes,
  mainRoutes: mainRoutes,
  error: errorRoutes,
}

export const useRouteList = () => ({
  allRoutes,
  authRoutes,
  mainRoutes,
  errorRoutes,
})
