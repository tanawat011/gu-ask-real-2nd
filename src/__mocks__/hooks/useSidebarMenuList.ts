export const useSidebarMenuList = () => {
  return [
    [
      {
        title: 'Dashboard',
        icon: 'dashboard',
        path: '/dashboard',
      },
      {
        title: 'Users',
        icon: 'user',
        path: '/users',
      },
      {
        title: 'Posts',
        icon: 'file',
        path: '/posts',
      },
    ],
    () => void 0,
  ]
}
