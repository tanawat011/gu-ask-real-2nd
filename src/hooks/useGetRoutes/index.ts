export * from './types'
export * from './useGetRoutes'
