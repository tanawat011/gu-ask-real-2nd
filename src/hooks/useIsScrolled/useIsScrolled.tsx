import type { OutletContextProps, Scroll } from 'types'

import { useEffect, useState } from 'react'

import { useOutletContext } from 'react-router-dom'

export const useIsScrolled = () => {
  const { scroll } = useOutletContext<OutletContextProps>()
  const [isScrolled, setIsScrolled] = useState(false)

  const handleScroll = ({ top }: Scroll) => {
    setIsScrolled(top > 0)
  }

  useEffect(() => {
    handleScroll(scroll)
  }, [scroll])

  return isScrolled
}
