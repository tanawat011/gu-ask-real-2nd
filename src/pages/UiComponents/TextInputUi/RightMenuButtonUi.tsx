import tw, { styled } from 'twin.macro'

import { RightMenuContent } from 'components/PageUiComponent'

import { containerId, menuList } from './data'

type RightMenuButtonUiProps = {
  isScrolled: boolean
}

const TwContainer = styled.div(({ scrolled }: { scrolled: boolean }) => {
  return [tw`z-10`, scrolled && tw`fixed`]
})

const _menuList = Object.values(menuList).map((item) => item.name)

export const RightMenuButtonUi: React.FC<RightMenuButtonUiProps> = ({
  isScrolled,
}) => {
  return (
    <TwContainer scrolled={isScrolled}>
      <RightMenuContent
        containerId={containerId}
        title='TABLE OF CONTENT'
        menuList={_menuList}
      />
    </TwContainer>
  )
}
