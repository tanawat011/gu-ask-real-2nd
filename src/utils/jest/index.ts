export * from './renderWithProviders'
export * from './twColor'
