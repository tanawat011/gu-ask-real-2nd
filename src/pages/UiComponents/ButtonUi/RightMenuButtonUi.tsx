import type { Dimension } from 'hooks/useElementDimension'
import type { ButtonUiDimensions } from 'recoils/atoms'

import tw, { styled } from 'twin.macro'

import { RightMenuContent } from 'components/PageUiComponent'
import { useContentContext } from 'hooks/useContentContext'
import { useIsScrolled } from 'hooks/useIsScrolled'

const TwContainer = styled.div(({ scrolled }: { scrolled: boolean }) => {
  return [tw`z-10`, scrolled && tw`fixed`]
})

type RightMenuButtonUiProps = {
  pageDimension: Dimension
  dimension: ButtonUiDimensions
}

export const RightMenuButtonUi: React.FC<RightMenuButtonUiProps> = ({
  pageDimension,
  dimension,
}) => {
  const { scroll } = useContentContext()
  const isScrolled = useIsScrolled()

  const {
    head,
    variant,
    color,
    size,
    shape,
    disabled,
    icon,
    withIcon,
    loading,
    block,
  } = dimension

  const handlePositionY = (_dimension: Dimension) => {
    const more = 1
    const marginBottom = 32

    return _dimension.clientHeight + _dimension.offsetTop + marginBottom - more
  }

  return (
    <TwContainer scrolled={isScrolled}>
      <RightMenuContent
        title='TABLE OF CONTENT'
        currentScrollPosition={scroll.top}
        pageHeight={pageDimension.clientHeight}
        menuList={[
          { title: 'Variant', scroll: handlePositionY(head) },
          { title: 'Color', scroll: handlePositionY(variant) },
          { title: 'Size', scroll: handlePositionY(color) },
          { title: 'Shape', scroll: handlePositionY(size) },
          { title: 'Disabled', scroll: handlePositionY(shape) },
          { title: 'Icon', scroll: handlePositionY(disabled) },
          { title: 'With Icon', scroll: handlePositionY(icon) },
          { title: 'Loading', scroll: handlePositionY(withIcon) },
          { title: 'Block', scroll: handlePositionY(loading) },
          { title: 'API', scroll: handlePositionY(block) },
        ]}
      />
    </TwContainer>
  )
}
