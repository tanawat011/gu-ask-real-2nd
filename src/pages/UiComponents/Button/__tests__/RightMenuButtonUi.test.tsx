import { mocked } from 'jest-mock'

import { useIsScrolled } from 'hooks/useIsScrolled'
import { renderWithProviders } from 'utils/jest'

import { RightMenuButtonUi } from '../RightMenuButtonUi'

jest.mock('@fortawesome/react-fontawesome')
jest.mock('components/PageUiComponent', () => ({
  RightMenuContent: () => <div>RightMenuContent</div>,
}))
jest.mock('hooks/useIsScrolled')

describe('<RightMenuButtonUi />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(<RightMenuButtonUi />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders correctly and scrolled', () => {
    expect.assertions(1)

    mocked(useIsScrolled).mockReturnValue(true)

    const { asFragment } = renderWithProviders(<RightMenuButtonUi />)

    expect(asFragment()).toMatchSnapshot()
  })
})
