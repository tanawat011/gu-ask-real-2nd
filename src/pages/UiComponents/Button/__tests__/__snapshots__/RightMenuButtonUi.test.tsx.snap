// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightMenuButtonUi /> renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  z-index: 10;
}

<div
    class="c0"
  >
    <div>
      RightMenuContent
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RightMenuButtonUi /> renders correctly and scrolled 1`] = `
<DocumentFragment>
  .c0 {
  z-index: 10;
  position: fixed;
}

<div
    class="c0"
  >
    <div>
      RightMenuContent
    </div>
  </div>
</DocumentFragment>
`;
